### PR TITLE
CI e2e test automation and Android

### PR DESCRIPTION
**Pull request by timwright12**
Detox tests for Android in CI and QA automation to post to slack thread during daily QA builds

**From Dylan:**
Tickets created for iOS and Android fixing tests and turning slack feature on

https://github.com/department-of-veterans-affairs/va-mobile-app/actions/runs/6471323790

https://github.com/department-of-veterans-affairs/va-mobile-app/assets/87150991/8aab205d-0c04-4430-964b-5f361713faed

### CODE REVIEW

**Comment by timwright12** [comment]
@Sparowhawk can you get a passing CI run before calling this done? We have everything passing locally, so I don't think we can just say all the tests are broken

File: .github/workflows/e2e_android.yml
Comment (timwright12) [inline .github/workflows/e2e_android.yml:11]: Let's take this out until it actually passes. No need to ping slack or run cron just for it to fail

Comment (Sparowhawk) [reply .github/workflows/e2e_android.yml]: @timwright12 I have taken it out. However these tests are not passing locally. I confirmed with @alexandec and @rbontrager that a decent number for iOS and Android are broken. Also unfortunately once one test breaks it breaks all the tests that occur after it.

I have moved this back to draft and ticket https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6879 is created for fixing the broken e2e tests that I found

Comment (timwright12) [reply .github/workflows/e2e_android.yml]: Might want to hop on a zoom with @rbontrager to investigate your set up. Every test outside of appointments was passing locally at last check

File: .github/workflows/e2e_android.yml
Comment (timwright12) [inline .github/workflows/e2e_android.yml:109]: This is in a local JSON file so we should just be able to pull it in so there's no need to keep it in sync https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/.detoxrc.json#L28

File: .github/workflows/e2e_android.yml
Comment (Sparowhawk) [reply .github/workflows/e2e_android.yml]: @timwright12 We were able to update detox and the platform error is no longer occurring. The issue now is that locally we have a different resolution/screen size than what we can enable on the emulator/simulator. So @rbontrager is now adding some scrolling to get most of the remaining tests to pass. We also came across a problem when changing json mock data in the files as well that seems to be a detox issue.

File: .github/workflows/e2e_android.yml
Comment (Sparowhawk) [reply .github/workflows/e2e_android.yml]: added

Comment (Sparowhawk) [reply .github/workflows/e2e_android.yml:130]: after playing with this for a while I was able to extra the name from the json file, but then this block of code no longer runs with this error:

Creating AVD.
  /bin/sh -c \echo no | avdmanager create avd --force -n Pixel_4_XL_API_28 --abi 'default/x86_64' --package 'system-images;android-28;default;x86_64' --device 'pixel_4_xl'
  Error: Package path (-k) not specified. Valid system image paths are:
  system-images;android-28;default;x86_64
  null
  
  So reverting back to the hard coded way that works as I tried trimming white space and everything else that might be going on. The command from what I can tell in the logs is the exact same and hard coded works and fetching from the file does not

**Comment by Sparowhawk** [comment]
@timwright12 This is now ready for review, I was able to pull the name from the json file, however by doing so the commands for that code block failed even though they were the exact same as the hard coded value, so I resorted to going back to hard coding that value. 

I have created/updated the tickets for the 6 tests for iOS and the 8 tests for Android along with the 2 tickets to turn on the slack feature for Android and iOS respectively once they are working as this ticket is now going greatly over scope.

File: .github/workflows/e2e_android.yml
Comment (timwright12) [inline .github/workflows/e2e_android.yml:12]: Remove before merging

File: .github/workflows/e2e_android.yml
Comment (timwright12) [inline .github/workflows/e2e_android.yml:82]: Is this duplicative with reactivecircus/android-emulator-runner@v2?

File: .github/workflows/e2e_ios.yml
Comment (timwright12) [inline .github/workflows/e2e_ios.yml:12]: Remove before merging

File: VAMobile/.detoxrc.json
Comment (timwright12) [inline VAMobile/.detoxrc.json:30]: Confirm with QA that this change is OK
@rbontrager

File: .github/workflows/e2e_ios.yml
Comment (timwright12) [inline .github/workflows/e2e_ios.yml:24]: @Sparowhawk this stuff should be in a separate ticket/PR, we're not alerting Slack yet until all the runs pass. Can you break it out? 

Also shouldn't be any iOS changes in this PR since it's an android ticket

File: .github/workflows/e2e_ios.yml
Comment (Sparowhawk) [reply .github/workflows/e2e_ios.yml]: removed

File: .github/workflows/e2e_android.yml
Comment (Sparowhawk) [reply .github/workflows/e2e_android.yml]: removed

File: .github/workflows/e2e_ios.yml
Comment (Sparowhawk) [reply .github/workflows/e2e_ios.yml]: removed the iOS changes

**Comment by timwright12** [comment]
Did we get a passing run in CI? Doesn't have to be all the tests, but at least 1 suite passing so we know we did it correctly

**Comment by Sparowhawk** [comment]
> Did we get a passing run in CI? Doesn't have to be all the tests, but at least 1 suite passing so we know we did it correctly

@timwright12 Yes we did, several individual tests passed. The ones that didn't have tickets for and are likely fixed by Rachel's iOS fixes in her other branch

**Comment by timwright12** [comment]
Can you link up to the workflow that passed in the description so we have that artifact?

**Comment by Sparowhawk** [comment]
@timwright12 added link and video of the passing and failing e2e tests with the workflow

**Comment by timwright12** [comment]
thanks, sounds good. I cant approve this because it was my PR originally. Whenever @rbontrager is comfortable, I'm good

**Review by Sparowhawk — APPROVED** [review]

**Pull request merged**
